eclass/kernel-build.eclass

```diff
@@ -225,6 +233,37 @@ kernel-build_src_configure() {
 	emake O="${WORKDIR}"/modprep "${MAKEARGS[@]}" olddefconfig
 	emake O="${WORKDIR}"/modprep "${MAKEARGS[@]}" modules_prepare
 	cp -pR "${WORKDIR}"/modprep "${WORKDIR}"/build || die
+
+	# Now that we have a release file, set KV_FULL
+	local relfile=${WORKDIR}/build/include/config/kernel.release
+	if [[ -z ${KV_FULL} ]]; then
+		KV_FULL=$(<"${relfile}") || die
+	fi
+
+	# Make sure we are about to build the correct kernel
+	if [[ ${PV} != *9999 ]]; then
+		local expected_ver=$(dist-kernel_PV_to_KV "${PV}")
+		local expected_rel=$(<"${relfile}")
+
+		if [[ ${KV_FULL} != ${expected_rel} ]]; then
+			eerror "Kernel release mismatch!"
+			eerror "Got:      ${KV_FULL}"
+			eerror "Expected: ${expected_rel}"
+			eerror "\$\{KV_FULL\} must match the kernel release."
+			die "Kernel release mismatch: got ${KV_FULL}, expected ${expected_rel}"
+		fi
+
+		if [[ ${KV_FULL} != ${expected_ver}* ]]; then
+			eerror "Kernel version mismatch!"
+			eerror "Got:      ${KV_FULL}"
+			eerror "Expected: ${expected_ver}*"
+			eerror "Kernel kernel version must match PV."
+			eerror "Please ensure you are applying the correct"
+			eerror "patches and that PV matches the prepared"
+			eerror "kernel's version."
+			die "Kernel version mismatch: got ${KV_FULL}, expected ${expected_ver}"
```

Comment:
```suggestion
			eerror "Kernel version does not match PV!"
			eerror "Source version: ${KV_FULL}"
			eerror "Expected (PV*): ${expected_ver}*"
			eerror "Please ensure that you are applying the correct patchset."
			die "Kernel version mismatch: got ${KV_FULL}, expected ${expected_ver}*"
```

Reply:
I don't think this one is an improvement at all and I specifically avoided copying the existing error message because I found it obtuse and confusing when preparing the asahi-kernel ebuild in the first place. I don't see how the message I have written is less clear.

Reply:
For a start, it is talking of "kernel version" and "kernel release" as if it were two different things. Also, it doesn't give you any hint what you need to change.